app/controllers/application_controller.rb

```diff
@@ -3,8 +3,161 @@
 class ApplicationController < Sinatra::Base
 
   configure do
+  	enable :sessions 
+  	set :session_secret, "jbsalt"
     set :public_folder, 'public'
     set :views, 'app/views'
   end
 
-end
\ No newline at end of file
+  get '/' do 
+  	erb :'index'
+  end 
+
+  get '/signup' do 
+  	if logged_in?
+  		redirect to '/tweets'
+  	else
+  		erb :'users/create_user' 
+  	end 
+  end 
+
+  post '/signup' do 
+  	# binding.pry
+  	if !params[:username].empty? && !params[:email].empty? && !params[:password].empty?
+	  	@user = User.create(username: params[:username], email: params[:email], password: params[:password])
+	  	session[:id] = @user.id 
+	  	redirect to "/tweets"
+	  else 
+	  	redirect to '/signup'
+	  end 
+  end 
+
+  get '/login' do 
+  	if !logged_in?
+  		erb :'users/login' 
+  	else
+  		redirect to '/tweets'
+  	end 
+  end 
+
+  post '/login' do 
+  	# binding.pry
+  	if logged_in? 
+  		redirect to '/tweets'
+  	else 
+	  	@user = User.find_by(username: params[:username])
+	  	if @user && @user.authenticate(params[:password])
+	  		session[:id] = @user.id 
+	  		redirect to "/tweets"
+	  	else 
+	  		redirect to '/login'
+	  	end 
+	  end 
+  end 
+
+  get '/users/:slug' do 
+  	@user = User.find_by_slug(params[:slug])
+  	erb :'/users/show'
+  end 
+
+  get '/logout' do 
+  	if logged_in?
+  		session.clear
+  	end 
+  	redirect to '/login' 
+  end 
+
+  get '/tweets' do 
+  	if logged_in?
+  		@user = User.find(session[:id])
+  		erb :'/tweets/tweets' 
+  	else
+  		redirect to '/login'
+  	end 
+  end 
+
+
+  get '/tweets/new' do 
+  	# binding.pry
+  	if logged_in?
+  		@user = User.find(session[:id])
+  		erb :'/tweets/create_tweet' 
+  	else
+  		redirect to '/login'
+  	end 
+  end 
+
+  post '/tweets' do 
+  	# binding.pry
+  	if !params[:content].empty?
+	  	@tweet = Tweet.create(content: params[:content])
```

Comment:
The tweet creation and user association can both be accomplished in the same step:

``` ruby
@tweet = Tweet.create(content: params[:content], user: current_user)
```